### ISSUE TITLE
Refactoring data using lookup

### ISSUE THREAD

**Issue by polc1410** [StatsGary/NHSDataDictionaRy#1]
I've opened a Q in Stackoverflow in case there is a simpler way to do this that I've missed:

https://stackoverflow.com/questions/66737678/am-i-missing-some-slicker-way-re-factoring-a-tibble-using-a-tibble-from-nhsdata

Basically I have some coded data which uses NHS DataDictionary code lists that it would be better to show as factors. 

Am I taking the best approach? And if so, would it be better for a future release to add the function (s)

Happy to try and submit pull requests etc. But thought I should ask first.

**Comment by StatsGary** [comment]
Yes, if you want to add a pull request, then I am happy to add your function as a collaborator.

Thanks,

Gary

Gary Hutson
________________________________
From: polc1410 ***@***.***>
Sent: 21 March 2021 21:27
To: StatsGary/NHSDataDictionaRy ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [StatsGary/NHSDataDictionaRy] Refactoring data using lookup (#1)


I've opened a Q in Stackoverflow in case there is a simpler way to do this that I've missed:

https://stackoverflow.com/questions/66737678/am-i-missing-some-slicker-way-re-factoring-a-tibble-using-a-tibble-from-nhsdata

Basically I have some coded data which uses NHS DataDictionary code lists that it would be better to show as factors.

Am I taking the best approach? And if so, would it be better for a future release to add the function (s)

Happy to try and submit pull requests etc. But thought I should ask first.

**Comment by polc1410** [comment]
OK. I will take a proper look at the code including error trapping etc.

**Comment by StatsGary** [comment]
@polc1410 have you had time to do this, or shall I close down?

**Comment by polc1410** [comment]
I've done the first function 

Will probably do the second over the weekend and then pop a pull request through.

**Comment by polc1410** [comment]
I see OpenPrescribing have published their code lists so I may also test if it works to import them. OP is not NHS but uses NHS data.

**Comment by polc1410** [comment]
I've sent a pull request through for the original extras.

I will work on a "OpenSafely" addition at some stage.  I need to understand how their user contributed lists work.  I'll open an issue for it when I have a clearer idea what it would do.
